sklearn/model_selection/_search.py

```diff
@@ -912,26 +923,34 @@ class GridSearchCV(BaseSearchCV):
             ``cv`` default value if None will change from 3-fold to 5-fold
             in v0.22.
 
-    refit : boolean, or string, default=True
+    refit : boolean, string, or callable, default=True
         Refit an estimator using the best found parameters on the whole
         dataset.
 
         For multiple metric evaluation, this needs to be a string denoting the
         scorer is used to find the best parameters for refitting the estimator
         at the end.
 
+        Where there are considerations other than maximum score in
+        choosing a best estimator, ``refit`` can be set to a function which
+        returns the selected ``best_index_`` given ``cv_results_``.
+
         The refitted estimator is made available at the ``best_estimator_``
         attribute and permits using ``predict`` directly on this
         ``GridSearchCV`` instance.
 
         Also for multiple metric evaluation, the attributes ``best_index_``,
         ``best_score_`` and ``best_params_`` will only be available if
         ``refit`` is set and all of them will be determined w.r.t this specific
-        scorer.
+        scorer. ``best_score_`` is not returned if refit is callable.
 
         See ``scoring`` parameter to know more about multiple metric
         evaluation.
 
+        .. versionadded:: 0.20
+            GridSearchCV supports ``refit`` = callable to add flexibility in
```

Comment:
Don't mention GridSearchCV here. Simply say "Support for callable added." the rest is documented above.